packages/rmdx/src/main/process.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright IBM Corp. 2022, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { createProcessor } from '@mdx-js/mdx'
+import remarkGfm from 'remark-gfm'
+import { visit, Visitor } from 'unist-util-visit'
+import { VFile } from 'vfile'
+import { matter } from 'vfile-matter'
+import { VFileMessage } from 'vfile-message'
+
+import { InputTooLargeException } from './exceptions/input-too-large-exception.js'
+import { MalformedMdxException } from './exceptions/malformed-mdx-exception.js'
+import { NoSuchHandlerException } from './exceptions/no-such-handler-exception.js'
+import { ProcessingException } from './exceptions/processing-exception.js'
+import {
+  AllowedComponents,
+  AstNode,
+  NodeHandler,
+  ParentAstNode,
+  ProcessedMdx,
+  RenderableAstNode
+} from './interfaces.js'
+import * as nodeHandlers from './node-handlers/index.js'
+
+const MAX_INPUT_LENGTH = 1000000 // bytes
+
+type NodeHandlers = Record<keyof typeof nodeHandlers, NodeHandler>
+
+const processor = createProcessor({
+  remarkPlugins: [remarkGfm]
+})
+
+/**
+ * The actual top-level visitor/node handler.
+ *
+ * @param data kwArgs input data
```

Comment:
is `kwArgs` here intentional? 😅

Reply:
Yeah, I can expand out that abbreviation though to "keyword args" or something more specific.

Reply:
Done!